src/lib/ab.ts

```diff
@@ -0,0 +1,44 @@
+import { Test, Variant } from './variants';
+
+const maxMvt = 1000000;
+
+/**
+ * If controlProportion is set then we use this to define the range of mvt values for the control variant.
+ * Otherwise we evenly distribute all variants across maxMvt.
+ *
+ * If controlProportion is set then the start of the mvt range for the control is derived using a hash of the test name.
+ * This is to avoid always putting the same users in the control for each test.
+ *
+ * So:
+ * controlProportion is in [0, 1]
+ * seed is in [0, maxMvt]
+ * controlRange is [seed, (seed + (maxMvt * controlProportion)) % maxMvt]
+ */
+
+export const getSeed = (name: string): number => {
+    let hash = 0;
+    if (name.length == 0) {
+        return hash;
+    }
+    for (let i = 0; i < name.length; i++) {
+        const char = name.charCodeAt(i);
+        hash = (hash << 5) - hash + char;
+        hash = hash & hash; // Convert to 32bit integer
+    }
```

Comment:
Ideally this algorithm would produce numbers that are evenly distributed between 0 and mvtMax - 1, do we know how true that is?

Reply:
that's a good point, and I don't think it can make that guarantee

Reply:
perhaps this is something that we could just generate once, e.g. in the tool, which is random in that range, and then is stored in the json with the rest of the test data

Reply:
I originally wanted to keep that logic/data out of the tool, but now I think you're absolutely right.
The tool can randomly generate a seed value _once_, and never update it after that.
Then `selectVariant` here uses the `seed` for the control mvt range. And no hashing required :)
Let's do that 👍

Reply:
Also, I'm not sure 'seed' is really the right term here, but I'm not sure what else it should be called!

Reply:
Yeah it's neat if all the logic could be here, but it might just make our lives easier to have it generated in the tool! Yeah I guess 'seed' sounds like something that should be fed into a random number generator. I guess as we have `controlProportion`, we could have `controlProportionOffset`? It's not as elegant as `seed` but maybe it does the job?

Reply:
👍 so we add to the EpicTest model:
```
interface ControlProportionSettings {
  controlProportion: number;
  controlProportionOffset: number;
}

interface EpicTest {
  ...
  controlProportionSettings?: ControlProportionSettings;
}
```

Reply:
I like it! 👍